databricks/koalas/mlflow.py

```diff
@@ -0,0 +1,166 @@
+    predict_type : a python basic type, a numpy basic type, a Spark type or 'infer'.
+       This is the return type that is expected when calling the predict function of the model.
+       If 'infer' is specified, the wrapper will attempt to determine automatically the return type
+       based on the model type.
+
+    Returns
+    -------
+    PythonModelWrapper
+        A wrapper around MLflow PythonModel objects. This wrapper is expected to adhere to the
+        interface of mlflow.pyfunc.PythonModel.
+
+    Examples
+    --------
+    Here is a full example that creates a model with scikit-learn and saves the model with
+     MLflow. The model is then loaded as a predictor that can be applied on a Koalas
+     dataframe.
+
+    We first initialize our MLflow environment:
+
+    >>> from mlflow.tracking import MlflowClient, set_tracking_uri
+    >>> import mlflow.sklearn
+    >>> from tempfile import mkdtemp
+    >>> d = mkdtemp("koalas_mlflow")
+    >>> set_tracking_uri("file:%s"%d)
+    >>> client = MlflowClient()
+    >>> exp = mlflow.create_experiment("my_experiment")
+    >>> mlflow.set_experiment("my_experiment")
+
+    We aim at learning this numerical function using a simple linear regressor.
+
+    >>> from sklearn.linear_model import LinearRegression
+    >>> train = pd.DataFrame({"x1": np.arange(8), "x2": np.arange(8)**2,
+    ...                       "y": np.log(2 + np.arange(8))})
+    >>> train_x = train[["x1", "x2"]]
+    >>> train_y = train[["y"]]
+    >>> with mlflow.start_run():
+    ...     lr = LinearRegression()
+    ...     lr.fit(train_x, train_y)
+    ...     mlflow.sklearn.log_model(lr, "model")
+    LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)
+
+    Now that our model is logged using MLflow, we load it back and apply it on a Koalas dataframe:
+
+    >>> from databricks.koalas.mlflow import load_model
+    >>> run_info = client.list_run_infos(exp)[-1]
+    >>> model = load_model("model", run_id = run_info.run_uuid)
+    >>> prediction_df = ks.DataFrame({"x1": [2.0], "x2": [4.0]})
+    >>> prediction_df["prediction"] = model.predict(prediction_df)
+    >>> prediction_df
+        x1   x2  prediction
+    0  2.0  4.0    1.355551
+
+    The model also works on pandas DataFrames as expected:
+
+    >>> model.predict(prediction_df[["x1", "x2"]].toPandas())
+    array([[1.35555142]])
+
+    Notes
+    -----
+    Currently, the model prediction can only be merged back with the existing dataframe.
+    Other columns have to be manually joined.
+    For example, this code will not work:
+
+    >>> df = ks.DataFrame({"x1": [2.0], "x2": [3.0], "z": [-1]})
+    >>> features = ["x1", "x2"]
+    >>> y = model.predict(df[features])
+    >>> # Will fail with a message about dataframes not aligned.
+    >>> df["y"] = y   # doctest: +SKIP
+
+    This is being tracked in the issue ticket https://github.com/databricks/koalas/issues/354.
```

Comment:
are we actually going to address this issue? if not, perhaps we should create a function that simply adds a column to an existing dataframe. Then it is more obvious what can be done.

Reply:
This is being resolved as of https://github.com/databricks/koalas/pull/381 . Then users will have to  `.merge()` for the time being, but this should be acceptable enough for most use cases. I will update the documentation once that PR is merged.

Reply:
Documentation updated.